audio-format-service/src/main.py

```diff
@@ -0,0 +1,361 @@
+    format: Optional[str] = Field(None, description="Detected format")
+    message: Optional[str] = Field(None, description="Validation message")
+
+
+class FormatDetectionResponse(BaseModel):
+    """Response model for format detection."""
+
+    format: str = Field(..., description="Detected audio format")
+    confidence: float = Field(..., description="Detection confidence (0.0-1.0)")
+
+
+class AudioMetadataResponse(BaseModel):
+    """Response model for audio metadata."""
+
+    duration: float = Field(..., description="Duration in seconds")
+    sample_rate: int = Field(..., description="Sample rate in Hz")
+    channels: int = Field(..., description="Number of audio channels")
+    bitrate: Optional[int] = Field(None, description="Bitrate in kbps")
+    format: str = Field(..., description="Audio format")
+    size: int = Field(..., description="File size in bytes")
+
+
+class ConversionResponse(BaseModel):
+    """Response model for format conversion."""
+
+    success: bool = Field(..., description="Whether conversion was successful")
+    original_format: str = Field(..., description="Original audio format")
+    target_format: str = Field(..., description="Target audio format")
+    file_path: Optional[str] = Field(None, description="Path to converted file")
+    message: Optional[str] = Field(None, description="Conversion message")
+
+
+class HealthResponse(BaseModel):
+    """Response model for health check."""
+
+    status: str = Field(..., description="Service status")
+    version: str = Field(..., description="Service version")
+    dependencies: Dict[str, str] = Field(..., description="Dependency status")
+
+
+# Global handler instance
+audio_handler: Optional[AudioFormatHandler] = None
+
+
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+    """Application lifespan manager."""
+    global audio_handler
+
+    # Initialize audio format handler
+    # Use mock handler for now - will be replaced with real implementation
+    audio_handler = MockAudioFormatHandler()
+
+    yield
+
+    # Cleanup
+    audio_handler = None
+
+
+# Create FastAPI application
+app = FastAPI(
+    title="Audio Format Service",
+    description="Service for audio format validation, detection, and conversion",
+    version="1.0.0",
+    lifespan=lifespan,
+)
+
+
+def get_audio_handler() -> AudioFormatHandler:
+    """Dependency to get the audio format handler."""
+    if audio_handler is None:
+        raise HTTPException(
+            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
+            detail="Audio format handler not initialized"
+        )
+    return audio_handler
+
+
+@app.get("/health", response_model=HealthResponse)
+async def health_check():
+    """Health check endpoint."""
+    return HealthResponse(
+        status="healthy",
+        version="1.0.0",
+        dependencies={
+            "audio_handler": "initialized" if audio_handler else "not_initialized",
```

Comment:
Should we follow the same pattern which we used below here instead of trying access to the global variable?:
```
handler: AudioFormatHandler = Depends(get_audio_handler)
```